docs/faq/the-self-parameter.md

```diff
@@ -0,0 +1,255 @@
+---
+title: The self parameter
+---
+
+import { SideBySide } from "@site/src/components/SideBySide";
+
+## Why is it there by default
+
+All functions, by default, have a `self` parameter to preserve JavaScript behaviour.
+
+Telling TypeScript not to allow `this` to be used will eliminate that parameter from its containing context.
+
+Other workarounds are available for removing this too.
+
+## Removing it
+
+### `noImplicitSelf`
```

Comment:
I think we should leave it as the last option, since more likely users would want to disable this behavior for Lua declarations, which are generally not affected by `noImplicitSelf`